src/icos_config/etherdelta.js

```diff
@@ -0,0 +1,69 @@
+import { toPromise } from '../utils';
+import { convertWeb3Value } from '../utils/web3';
+
+export default {
+  crowdSaleTokenContract: '0xCe53a179047ebed80261689367c093C90A94cC08',
+  information: {
+    name: 'etherdelta',
+    website: 'https://etherdelta.com/',
+    logo: 'https://pbs.twimg.com/profile_images/775077934650494976/YuV0wuPn_normal.jpg',
+  },
+  events: {
+    Buy: {
+      args: {
+        tokens: 'token',
+        ether: 'eth',
+        sender: 'sender',
+      },
+      firstTransactionBlockNumber: 4830751,
+      lastTransactionBlockNumber: null,
+      maxBlocksInChunk: 42960,
+      countTransactions: true,
+    },
+  },
+  icoParameters: {
+    cap: async (web3, icoContract) =>
+      ['Hard Cap: 750M EDT'], // from website    
+    startDate: async (web3, icoContract) => {
+      const openTime = await toPromise(icoContract.openTime)();
+      return convertWeb3Value(openTime.valueOf(), 'timestamp').formatDate();
+    },
+    endDate: async (web3, icoContract) => {
+      const closeTime = await toPromise(icoContract.closeTime)();
+      return convertWeb3Value(closeTime.valueOf(), 'timestamp').formatDate();
+    },
+    status: async (web3, icoContract) => {
+      const now = Math.floor(new Date().getTime() / 1000);
```

Comment:
why not to use `getPeriod` that does exactly the same and convert uint to string with state names?